connect/src/com/telenor/connect/SdkProfile.java

```diff
@@ -24,10 +24,23 @@
     WellKnownAPI.WellKnownConfig getWellKnownConfig();
     boolean isInitialized();
 
-    void onStartAuthorization(OnStartAuthorizationCallback callback);
+    void initializeFromUri(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            Uri initFrom,
+            OnDeliverAuthorizationUriCallback callback);
 
-    interface OnStartAuthorizationCallback {
-        void onSuccess();
+    void onStartAuthorization(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            OnStartAuthorizationCallback callback);
+
+    interface OnDeliverAuthorizationUriCallback {
+        void onSuccess(Uri authorizationStartUri);
         void onError();
     }
+
+    interface OnStartAuthorizationCallback extends OnDeliverAuthorizationUriCallback {
```

Comment:
Any reason why having this is the same hierarchy? Why isn't there a completely separate callback for mobile connect for the operator discovery result (irregardless of the result was delivered via the manual selection or automatically)? It looks weird to have both kinds of callbacks in the same class (eg, sometimes you implement the onDivert method by throwing RuntimeException).